### PR TITLE
👔 Implement GET "/weeks/{weekId}/dashboard"

### PR DESCRIPTION
**Pull request by CalculusEnjoyer**
Implement GET "/weeks/{weekId}/dashboard" and write tests for CandidateService methods that were added in order to implement this endpoint.

### CODE REVIEW

**Comment by CalculusEnjoyer** [comment]
|File|Coverage [41.97%]|:green_apple:|
|:-|:-:|:-:|
|[CandidateSlotValidator.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fvalidation%2FCandidateSlotValidator.java)|100%|:green_apple:|
|[UtilValidator.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fvalidation%2FUtilValidator.java)|100%|:green_apple:|
|[BookingNotFoundException.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fexceptions%2FBookingNotFoundException.java)|100%|:green_apple:|
|[BookingService.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingService.java)|75.44%|:green_apple:|
|[WeekService.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FWeekService.java)|66.15%|:green_apple:|
|[InterviewerService.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FInterviewerService.java)|64.63%|:green_apple:|
|[CoordinatorController.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FCoordinatorController.java)|63.52%|:green_apple:|
|[InterviewerDto.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2Fdto%2FInterviewerDto.java)|53.13%|:green_apple:|
|[InterviewerSlotDto.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2Fdto%2FInterviewerSlotDto.java)|48.53%|:green_apple:|
|[CandidateSlotDto.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2Fdto%2FCandidateSlotDto.java)|47.73%|:green_apple:|
|[Booking.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2FBooking.java)|42.45%|:green_apple:|
|[CandidateService.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FCandidateService.java)|40.35%|:green_apple:|
|[UserService.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FUserService.java)|20.08%|:green_apple:|
|[InterviewApplicationExceptionHandler.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fexceptions%2FInterviewApplicationExceptionHandler.java)|6.25%|:x:|
|[InterviewerValidator.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fvalidation%2FInterviewerValidator.java)|0%|:x:|
|[DashboardDayDto.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/14d3d703170f610616ca9e4be74860661f7c61ff/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2Fdto%2FDashboardDayDto.java)|0%|:x:|

|Total Project Coverage|43.35%|:green_apple:|
|:-|:-:|:-:|

**Comment by VladOSIO2** [comment]
Everything is good, only some grammar mistakes in javadocs, mostly where "booking" was changed to "bookings"

image

File: src/main/java/com/intellias/intellistart/interviewplanning/services/CandidateService.java [hunk 1]
```diff
@@ -99,8 +106,28 @@ public Candidate registerCandidate(Candidate candidate) {
...
+            || weekService.getWeekNumFrom(slot.getDateTo().toLocalDate()) == weekNum)
+        .collect(Collectors.toList());
   }
 
+  /**
+   * Gets all candidate slots that are arranged on a particular week and day.
+   */
+  public List<CandidateSlot> getCandidateSlotsForWeekAndDayOfWeek(int weekNum, int dayOfWeek) {
+    return slotRepository.findAll().stream()
+        .filter(slot -> weekService.getWeekNumFrom(slot.getDateFrom().toLocalDate()) == weekNum
```
Comment (VladOSIO2) [inline src/main/java/com/intellias/intellistart/interviewplanning/services/CandidateService.java:128]: That's a bit complicated, to be honest). I'd ask you to decompose this, for example moving the inner lambdas into separate private static methods for a better readability

**Review by VladOSIO2 — APPROVED** [review]
Approved, the code does it's work well, I would only ask you to make a little refactoring, described in the comments, for a better code readability

**Comment by CalculusEnjoyer** [comment]
|File|Coverage [41.93%]|:green_apple:|
|:-|:-:|:-:|
|[CandidateSlotValidator.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fvalidation%2FCandidateSlotValidator.java)|100%|:green_apple:|
|[UtilValidator.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fvalidation%2FUtilValidator.java)|100%|:green_apple:|
|[BookingNotFoundException.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fexceptions%2FBookingNotFoundException.java)|100%|:green_apple:|
|[BookingService.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingService.java)|75.44%|:green_apple:|
|[WeekService.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FWeekService.java)|66.15%|:green_apple:|
|[InterviewerService.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FInterviewerService.java)|64.63%|:green_apple:|
|[CoordinatorController.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FCoordinatorController.java)|63.52%|:green_apple:|
|[InterviewerDto.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2Fdto%2FInterviewerDto.java)|53.13%|:green_apple:|
|[InterviewerSlotDto.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2Fdto%2FInterviewerSlotDto.java)|48.53%|:green_apple:|
|[CandidateSlotDto.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2Fdto%2FCandidateSlotDto.java)|47.73%|:green_apple:|
|[CandidateService.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FCandidateService.java)|40.35%|:green_apple:|
|[UserService.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FUserService.java)|20.08%|:green_apple:|
|[InterviewApplicationExceptionHandler.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fexceptions%2FInterviewApplicationExceptionHandler.java)|6.25%|:x:|
|[InterviewerValidator.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2Fvalidation%2FInterviewerValidator.java)|0%|:x:|
|[DashboardDayDto.java](https://github.com/CalculusEnjoyer/intellistart-java-2022-vanquishers/blob/fe424553def7b7e9f7ebb11ba214b7a1646b1e7d/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2Fdto%2FDashboardDayDto.java)|0%|:x:|

|Total Project Coverage|43.35%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/intellias/intellistart/interviewplanning/services/CandidateService.java (hunk 1 above)
Comment (CalculusEnjoyer) [reply src/main/java/com/intellias/intellistart/interviewplanning/services/CandidateService.java]: Oh, really better to separate it into multiple methods.
I have refactored it and have fixed typos in Booking.

**Pull request merged**